### PR TITLE
complete google login

### PR DESCRIPTION
**Pull request by TauJiang**

### CODE REVIEW

File: vwb/Gemfile [hunk 1]
```diff
@@ -3,6 +3,13 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.7.0'
 
+gem 'devise', github: 'heartcombo/devise', branch: 'ca-omniauth-2'
+gem 'omniauth', '~>1.9.1'
+gem 'omniauth-google-oauth2'
+gem 'omniauth-facebook'
+gem 'omniauth-github'
+gem 'figaro'
+
```
Comment (dantp0929) [inline vwb/Gemfile:12]: Gems for google oauth looks correct!

File: vwb/db/schema.rb [hunk 2]
```diff
@@ -35,6 +35,16 @@
...
 
+  create_table "userlogins", force: :cascade do |t|
+    t.string "email", null: false
+    t.string "full_name"
+    t.string "uid"
+    t.string "avatar_url"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["email"], name: "index_userlogins_on_email", unique: true
+  end
```
Comment (dantp0929) [inline vwb/db/schema.rb:46]: data looks correct.

**Review by dantp0929 — APPROVED** [review]
Looks good, should be able to merge.

File: vwb/Gemfile (hunk 1 above)
Comment (bryalara) [reply vwb/Gemfile:12]: gem files look correct and essential for building login function

File: vwb/db/schema.rb (hunk 2 above)
Comment (bryalara) [reply vwb/db/schema.rb:46]: schema value look right, good job on making sure emails are unique

**Pull request merged**